api/cluster.go

```diff
@@ -505,3 +508,28 @@ func (s *Server) peerQuerySpeculativeChan(ctx context.Context, data cluster.Trac
 
 	return resultChan, errorChan
 }
+
+func (s *Server) indexMetaTagRecordUpsert(ctx *middleware.Context, req models.IndexMetaTagRecordUpsert) {
+	if s.MetricIndex == nil {
+		response.Write(ctx, response.NewMsgp(200, &models.MetaTagRecordUpsertResult{}))
+		return
+	}
+
+	record := idx.MetaTagRecord{
+		MetaTags: req.MetaTags,
+		Queries:  req.Queries,
+	}
+
+	result, created, err := s.MetricIndex.MetaTagRecordUpsert(req.OrgId, record)
+	if err != nil {
+		response.Write(ctx, response.NewError(http.StatusBadRequest, err.Error()))
+		return
+	}
+
+	response.Write(ctx, response.NewMsgp(200, &models.MetaTagRecordUpsertResult{
+		MetaTags: result.MetaTags,
+		Queries:  result.Queries,
+		ID:       result.ID,
```

Comment:
isn't the id an in-process implementation detail that we don't want to expose externally (so that we can change it as desired)

Reply:
that's true, I can't think of a reason why the user would need to know it. I'll change it so it doesn't get exposed